Source/WarpX.cpp

```diff
@@ -695,64 +716,50 @@ WarpX::ReadParameters ()
         pp_interpolation.query("noz", noz);
 
 #ifdef WARPX_USE_PSATD
+
         if (WarpX::maxwell_solver_id == MaxwellSolverAlgo::PSATD) {
+
             // For momentum-conserving field gathering, read from input the order of
             // interpolation from the staggered positions to the grid nodes
-            if (field_gathering_algo == GatheringAlgo::MomentumConserving) {
-                pp_interpolation.query("field_gathering_nox", field_gathering_nox);
-                pp_interpolation.query("field_gathering_noy", field_gathering_noy);
-                pp_interpolation.query("field_gathering_noz", field_gathering_noz);
+            if (WarpX::field_gathering_algo == GatheringAlgo::MomentumConserving) {
+                pp_interpolation.query("field_centering_nox", field_centering_nox);
+                pp_interpolation.query("field_centering_noy", field_centering_noy);
+                pp_interpolation.query("field_centering_noz", field_centering_noz);
             }
 
+            // Read order of finite-order centering of currents (nodal to staggered)
+            if (WarpX::do_current_centering) {
+                pp_interpolation.query("current_centering_nox", current_centering_nox);
+                pp_interpolation.query("current_centering_noy", current_centering_noy);
+                pp_interpolation.query("current_centering_noz", current_centering_noz);
+            }
+
+            // TODO Is centering of currents implemented with mesh refinement?
```

Comment:
Should we remove this line and open a corresponding Github issue?

Reply:
Yes, I can open an issue. For the moment, I removed the `TODO` comment and added an `amrex::Abort` to warn users that finite-order centering of currents is not implemented with mesh refinement: https://github.com/ECP-WarpX/WarpX/pull/1763/commits/24160ff90a9170f7561abcd75445cfc84a22e616.